microservices/validateApi/validator/validator.py

```diff
@@ -48,35 +48,81 @@ def validate(rule, resultObj):
 
 
 def read_file_and_evaluate(source, result):
-    file_resp, file_attributes = read_file(result['file_id'], not(source.find('${file.content}') == -1))
+    _, file_attributes = read_file(
+        result['file_id'], '${file.content}' in source)
     return evaluate_source(source, file_attributes)
 
 
 def evaluate_source(source, file_attributes):
-    munchified_attributes = munchify(file_attributes)
-
-    source = source.replace("${file.", "{0.")
-
+    PREFIX = "file."
     result = False
     message = ""
 
+    # def format_fn(val):
+    #     """
+    #     This function munchifies the file attributes and formats string val
+    #     :param val: A template string
+    #     :return: An interpolated string
+    #     """
+    #     val_temp = val.replace("${file.", "{0.")
+    #     return val_temp.format(munchify(file_attributes))
+
     try:
-        source = source.format(munchified_attributes)
-        old_stdout = sys.stdout
-        redirected_stdout = sys.stdout = StringIO()
-        exec(source)
-        sys.stdout = old_stdout
-        execOutput = redirected_stdout.getvalue().lower()
-        print(execOutput)
-        result = execOutput in ("yes", "true", "t", "1", "yes\n", "true\n", "t\n", "1\n")
+        munch_attr = munchify(file_attributes)
+
+        src_split = re.split(r"(\${)(file..*?)(})", source)
+        src_parts = [x for x in src_split if x not in ("${", "}")]
+        src_sub = [munch_attr.get(x[len(PREFIX):]) if x.startswith(
+            PREFIX) else x for x in src_parts]
+        exec_src = "".join(map(str, src_sub))
+        # exec_src = niño_cédille_postulate(source, format_fn)
+
+        exec_output = execute_script(exec_src)
+        log.info(exec_output)
+        result = exec_output.rstrip() in ("yes", "true", "t", "1")
     except (Exception, NameError) as e:
         log.error(e)
         message = str(e)
 
     return result, message
 
-def read_file(file_id, deep_read=False):
 
+# def niño_cédille_postulate(source, fn):
+#     """
+#     This function implements the Niño Cédille Postulate (NCP).
+#     NCP asserts that no strings will likely use both a niño and cédille as
+#     substitute characters to temporarily escape out curly braces.
+#     Note: this will not work on JSON sources with more than one nested level
+
+#     :param source: A string to be handled by NCP
+#     :param fn: A lambda function to be applied while string is NCP'ed
+#     :return: A string with NCP and fn applied if there is no error; else source
+#     """
+#     ncp_forward = re.compile(r"(?<!\$)(?:{)(.*?)(?:})")
+#     ncp_reverse = re.compile(r"(?<!\$)(?:ñ)(.*?)(?:ç)")
+
+#     result = ""
+#     try:
+#         ncp_source = ncp_forward.sub(r"ñ\1ç", source)
+#         fn_source = fn(ncp_source)
+#         result = ncp_reverse.sub(r"{\1}", fn_source)
+#     except (Exception, NameError) as e:
+#         log.error(e)
+#         result = source
+
+#     return result
+
+
```

Comment:
This was previously in the try catch block in the function because if STDERR is written to you won't catch it and will error out. Need to fail gracefully.

Reply:
Added a try except and rethrow pattern.